### PR TITLE
Add LinkedIn provider

### PR DESCRIPTION
**Pull request by robertd**
Credits go to @mkoskine for initial implementation in #64.
I just cleaned few things up...

Cheers!

### CODE REVIEW

File: lib/providers/linkedin.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+exports = module.exports = function (options) {
+    
+    return {
+        protocol: 'oauth2',
+        auth: 'https://www.linkedin.com/uas/oauth2/authorization',
+        token: 'https://www.linkedin.com/uas/oauth2/accessToken',
+        scope: ['r_fullprofile', 'r_emailaddress', 'r_contactinfo'],
+        scopeSeparator: ',',
+        profile: function (credentials, params, get, callback) {
+            var query = {
```
Comment (geek) [inline lib/providers/linkedin.js]: we have a different style guide.  I'll update the linting rules at some point... but please add a newline after every function declaration.  The reason is to make it easier to identify new scopes

File: lib/providers/linkedin.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                format: 'json',
+                appsecret_proof: Crypto.createHmac('sha256', this.clientSecret).update(credentials.token).digest('hex')
+            };
+            
+            get('https://api.linkedin.com/v1/people/~', query, function (profile) {
+                credentials.profile = {
```
Comment (geek) [inline lib/providers/linkedin.js]: please add a newline

**Comment by geek** [comment]
Looks good overall... just a couple of minor changes and need to pull the latest from master so it can be merged.  Thanks!

File: lib/providers/linkedin.js (hunk 1 above)
Comment (robertd) [inline lib/providers/linkedin.js]: will do

**Comment by robertd** [comment]
Let me know if there are any more styling discrepancies...

Thanks

**Comment by geek** [comment]
@robertd so fast!  Nicely done.

**Comment by robertd** [comment]
squashed too ... just in case :)

**Pull request merged**